webrender/src/frame.rs

```diff
@@ -67,15 +67,23 @@ impl NestedDisplayListInfo {
 
     fn convert_scroll_id_to_nested(&self, id: &ClipId) -> ClipId {
         if id.is_root_scroll_node() {
-            self.scroll_node_id
+            if id.pipeline_id() == self.scroll_node_id.pipeline_id() {
+                self.scroll_node_id
+            } else {
+                *id
+            }
         } else {
             self.convert_id_to_nested(id)
```

Comment:
I think that in this case it makes sense to treat ClipIds from other pipelines the same way, no matter if they are the root id or not. If you can guarantee that ids from this other pipeline will be unique across the entire display list, then I think we should just do an early return in these methods.